x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BasicTokenizer.java

```diff
@@ -101,38 +101,25 @@ public BasicTokenizer(boolean isLowerCase, boolean isTokenizeCjkChars) {
             // At this point text has been tokenized by whitespace
             // but one of the special never split tokens could be adjacent
             // to one or more punctuation characters.
-            if (isCommonPunctuation(token.codePointAt(token.length() - 1))) {
-                int lastNonPunctuationIndex = findLastNonPunctuationIndex(token);
-                if (lastNonPunctuationIndex >= 0 && neverSplit.contains(token.substring(0, lastNonPunctuationIndex + 1))) {
-                    processedTokens.add(token.substring(0, lastNonPunctuationIndex + 1));
-                    processedTokens.addAll(splitOnPunctuation(token.substring(lastNonPunctuationIndex + 1)));
-                    continue;
+            List<String> splitOnCommonTokens = splitOnPredicate(token, BasicTokenizer::isCommonPunctuation);
```

Comment:
It seems wrong to be treating "common" punctuation differently to punctuation here.  It means this method will work differently for a straight apostrophe and a curly apostrophe for example, which seems like something that will get raised as a bug in the future.

It was like this before this change too, so if it's difficult to fix then please at least open the bug now so that when somebody complains we can point them at the already opened issue instead of investigating from scratch at that time.

Reply:
https://github.com/elastic/elasticsearch/issues/80484